src/Host/Broker/Impl/ApiErrorResult.cs

```diff
@@ -1,35 +1,25 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-using System;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.R.Host.Protocol;
 
 namespace Microsoft.R.Host.Broker {
     internal sealed class ApiErrorResult : ObjectResult {
-        private BrokerApiError _brokerApiError;
-        private readonly string _message;
-         
-        public ApiErrorResult(BrokerApiError error, string message = null) : base(error) {
+        public ApiErrorResult(HttpResponse response, BrokerApiError error, string message = null) : base(error) {
             // https://tools.ietf.org/html/rfc7231#section-6.5.1
             // The 400(Bad Request) status code indicates that the server cannot or 
             // will not process the request due to something that is perceived to be 
             // a client error(e.g., malformed request syntax, invalid request message 
             // framing, or deceptive request routing).
             StatusCode = StatusCodes.Status400BadRequest;
 
-            _brokerApiError = error;
-            _message = message;
-         }
-
-        public override void ExecuteResult(ActionContext context) {
-            var headers = context.HttpContext.Response.Headers;
-            headers.Add(CustomHttpHeaders.RTVSApiError, new Extensions.Primitives.StringValues(_brokerApiError.ToString()));
-            if (!string.IsNullOrEmpty(_message)) {
-                headers.Add(CustomHttpHeaders.RTVSBrokerException, _message);
+            var headers = response.Headers;
+            headers.Add(CustomHttpHeaders.RTVSApiError, new Extensions.Primitives.StringValues(error.ToString()));
```

Comment:
Because I found that ExecuteResult actually never gets called so we ended up again without diagnostics. Anything I can do to make it executed?

---
In reply to: [82292942](https://github.com/Microsoft/RTVS/pull/2447#discussion_r82292942) [](ancestors = 82292942)